README.md

```diff
@@ -1,16 +1,171 @@
+argument_ which they will hand each of their elements off to. In the case of
+`map()` it hands each element to the function and captures the return value of
+the function into a new `Array`.
 
-## SWBAT 2
+This code is more expressive because it lives up to the promise of map. It
+creates a new `Array` after each element is "touched" by a function.
+
+One drawback to this code is that the `studentPatronusName` function doesn't do
+much work. It just returns something that the `student` _already_ knew how to
+do.  What if we use a function expression ("anonymous function") instead?
+
+### `map` and a function expression
+
+```js
+let students = [harryPotter, ronWeasley, hermioneGranger, ginevraWeasley];
+let patroni = students.map(function(student) {
+  return student.namePatronus();
+});
+//=> patroni = ["Stag", "Jack Russell Terrier", "Otter", "Horse"];
+```
+
+Wow, that's more terse! It has all the same advantages of the previous version.
+
+### `map` and an arrow function
+
+Thanks to arrow functions, we can shorten up the function expression to:
+
+```js
+// When the parameter list is only one element, we can drop () !
+let students = [harryPotter, ronWeasley, hermioneGranger, ginevraWeasley];
+let patroni = students.map( student => student.namePatronus() )
+//=> patroni = ["Stag", "Jack Russell Terrier", "Otter", "Horse"];
+```
+
+Wow, things are much less verbose! There is much less noisy JavaScript code so
+the _expressiveness_ has increased: "`patroni` is the result of `map`-ing
+`students` and asking each for the result of its `namePatronus` method".
+
+For the rest of of these examples, we'll use the arrow function.
+
+## Demonstrate `map()` with Complex Data Structures
+
+Let's use the `map()` function on a trickier data structure — a list of robots.
+To start things off, we have an array of robots. Now, let's activate all of
+them. An activated robot needs to be marked as such using the `isActivated`
+boolean, as well as have its number of modes doubled:
+
+```js
+const robots = [
+  { name: 'Johnny 5', modes: 5, isActivated: false, },
+  { name: 'C3PO', modes: 3, isActivated: false, },
+  { name: 'Sonny', modes: 2.5, isActivated: false, },
+  { name: 'Baymax', modes: 1.5, isActivated: false, },
+];
+
+const activatedRobots = robots.map(function (robot) {
+  return Object.assign({}, robot, {
+    modes: robot.strength * 2,
+    isActivated: true,
+  });
+});
+
+console.log(activatedRobots);
+
+/*
+ Result:
+ [
+   { name: 'Johnny 5', modes: 10, isActivated: true },
+   { name: 'C3PO', modes: 6, isActivated: true },
+   { name: 'Sonny', modes: 5, isActivated: true },
+   { name: 'Baymax', modes: 3, isActivated: true }
+ ]
+*/
+```
+
+With  the native `map()` function that is a property
+of `Array`'s prototype. It gives us the exact same result! Now that we know how
+map is implemented, it holds no more secrets for us! We can discard our own `map()`
+function and just use the `.map()` property on arrays. Sweet!
```

Comment:
This applies the entire main body of the text: it's all structured well and clearly expressed, but I think this is an occasion that our readability tool would be helpful. There are a lot of longer words in here that I wonder couldn't be switched out for simpler ones. I would recommend just going through with that in mind and seeing if anything can be simplified.